sequences.lisp

```diff
@@ -1935,3 +1935,184 @@ Repetitions that are not adjacent are left alone.
+                 (let ((result (make-array (+ length diff) :element-type
+                                           (array-element-type vector))))
+                   ;; Copy the beginning.
+                   (replace result vector
+                            :start1 0 :end1 start
+                            :start2 0 :end2 start)
+                   ;; Copy the new part.
+                   (when new (replace result new :start1 start))
+                   ;; Copy the end.
+                   (replace result vector
+                            :start1 (+ end diff) :end1 (length result)
+                            :start2 end :end2 length)
+                   result))
+                ((plusp diff)
+                 ;; The array is adjustable and we are making it longer.
+                 ;; Adjust it and only then move the tail.
+                 (adjust-array vector (+ length diff))
+                 (replace vector vector
+                          :start1 (+ end diff) :end1 (+ length diff)
+                          :start2 end :end2 length)
+                 (when new (replace vector new :start1 start))
+                 vector)
+                (t
+                 ;; The array is adjustable and we are making it shorter.
+                 ;; Move the tail and only then adjust it.
+                 (replace vector vector
+                          :start1 (+ end diff) :end1 (+ length diff)
+                          :start2 end :end2 length)
+                 (adjust-array vector (+ length diff))
+                 (when new (replace vector new :start1 start))
+                 vector))))))
+
+(defun splice-vector (vector new start end)
+  (declare (type vector vector))
+  (declare (type sequence new))
+  (declare (type alexandria:array-index start end))
+  (let ((diff-removed (- end start))
+        (diff-added (length new))
+        (vector (copy-seq vector)))
+    (if (= diff-removed diff-added)
+        (replace vector new :start1 start)
+        (let* ((diff (- diff-added diff-removed))
+               (length (length vector)))
+          (declare (type array-size-difference diff))
+          ;; We must copy and overwrite the whole array.
+          (let ((result (make-array (+ length diff) :element-type
+                                    (array-element-type vector))))
+            ;; Copy the beginning.
+            (replace result vector
+                     :start1 0 :end1 start
+                     :start2 0 :end2 start)
+            ;; Copy the new part.
+            (when new (replace result new :start1 start))
+            ;; Copy the end.
+            (replace result vector
+                     :start1 (+ end diff) :end1 (length result)
+                     :start2 end :end2 length)
+            result)))))
+
+;;; TODO: there is no extensible sequence support in SPLICE/NSPLICE right now.
+;;;       Should there be? It doesn't seem like Serapeum is tested with
+;;;       extensible sequences at all.
```

Comment:
I do try to keep extensible sequences in mind, but since they're not currently tested you don't have to worry about them here. But do leave the TODO in.